### DISCUSSION TITLE
What is Overfitting, and How Can You Avoid It?

### DISCUSSION THREAD

**Discussion by luffy-Monkey123** [luffy-Monkey123/python-cool-programs#14] [Q&A] [answered]
What is Overfitting, and How Can You Avoid It?

**Comment by RUTAGENGWA-ASANTE-BRUCE** [comment, marked as answer]
The Overfitting is a situation that occurs when a model learns the training set too well, taking up random fluctuations in the training data as concepts. These impact the model’s ability to generalize and don’t apply to new data. 

When a model is given the training data, it shows 100 percent accuracy—technically a slight loss. But, when we use the test data, there may be an error and low efficiency. This condition is known as overfitting.

There are multiple ways of avoiding overfitting, such as:

Regularization. It involves a cost term for the features involved with the objective function
Making a simple model. With lesser variables and parameters, the variance can be reduced 
Cross-validation methods like k-folds can also be used
If some model parameters are likely to cause overfitting, techniques for regularization like LASSO can be used that penalize these parameters
